tests/scripts/all.sh

```diff
@@ -924,10 +924,10 @@ component_test_default_cmake_gcc_asan () {
     tests/context-info.sh
 }
 
-component_test_full_cmake_gcc_asan () {
-    msg "build: full config, cmake, gcc, ASan"
+component_test_full_cmake_clang_asan () {
```

Comment:
There's a `component_test_full_cmake_clang` and it would seem natural to merge it with `component_test_full_cmake_clang_asan`. After all, we don't need to test both an asan and a non-instrumented build.

Unfortunately, the two components have different requirements. The whole point of this change is to insist on a recent clang. When running Ubuntu 20.04, that also means a recent openssl which doesn't support the same parts of `compat.sh`. Between the two components, all of `compat.sh` should run. But I don't think there's a single version of OpenSSL that supports everything (ARIA requires ≥1.1, null ciphersuites require &lt;1.1, and for 2.28 there's also DES).

This is more difficult than I'd hoped and I may revise my ambitions down for Travis, especially considering that it's kind of on its way out now that we have OpenCI.

Reply:
In today's team meeting we decided that we could remove this component from Travis. The main reason were keeping Travis in addition to Jenkins was that it has public logs, but now with OpenCI so does Jenkins. We aren't going to invest time into Travis at this point.

Reply:
Then again, having a run in the full config on Travis is really convenient. It lets us see at first glance if the “baseline” fails. So I've kept one, just not using `all.sh`, so it doesn't have to have the same expectations of completeness.

Since this run doesn't use our official openssl and gnutls versions, it is not exactly identical to even a subset of a Jenkins job. This was already the case before: while we use the same _versions_ of openssl and gnutls on Jenkins as in Ubuntu 16.04, we don't use the same _builds_, and this has bitten us before when Ubuntu removed some obsolete algorithms from the default configuration.

Having something on the CI that's close to a typical developer machine (at least closer than an out-of-support Ubuntu release) helps maintain the property that just running `make && make test && tests/ssl-opt.sh && tests/compat.sh` can be expected to work out of the box.

I hope that by the time we retire Travis, we'll have modernized our interoperability testing (specifically, version reliance in the SSL test scripts) so that Jenkins can ensure this property.